### PR TITLE
OCPBUGS-7076: Copy Day2 BMH if Agent is installing

### PR DESCRIPTION
**Pull request by CrystalChun**
[OCPBUGS-7076](https://issues.redhat.com/browse/OCPBUGS-7076)
Previously the BMAC would copy the BMH and 
Machine CRs of a Day2 worker to the spoke cluster before
the Agent began installing. If the Agent doesn't
start installing within the two hours these CRs
were copied over, the CSRs for these nodes will
not be approved by the machine approver pod.

To prevent this from happening, only copy
the BMH and Machine CRs to the spoke cluster
when the Agent has started installing.

## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [x] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

/cc @eranco74

### CODE REVIEW

File: internal/controller/controllers/bmh_agent_controller.go
Comment (avishayt) [inline internal/controller/controllers/bmh_agent_controller.go:979]: HostStatusInstalled should also be included, in case the controller somehow missed the installation (e.g., it was down or stuck)

**Comment by avishayt** [comment]
Why not wait until the node is fully installed?  What happens if the installation itself takes more than 2-3 hours (for example, a long time to pull images, or the host booted into the wrong disk and it took time to address)?

**Comment by CrystalChun** [comment]
> Why not wait until the node is fully installed? What happens if the installation itself takes more than 2-3 hours (for example, a long time to pull images, or the host booted into the wrong disk and it took time to address)?

@avishayt Thank you for the feedback! 

I've tested this out by only checking for `HostStatusInstalled`, as suggested, and observed that the Agent gets stuck in `Rebooting` with the state `installing-pending-user-action`. The install never completes.

The spoke cluster's `machine approver` pod errors out with
```
E0602 00:52:57.290085       1 csr_check.go:263] csr-xbfck: failed to find machine for node extraworker-1, cannot approve
I0602 00:52:57.290119       1 controller.go:233] csr-xbfck: CSR not authorized
```
where `extraworker-1` is my day 2 worker node that never gets installed.

Please correct me if I'm wrong, but it seems like copying the BMH & Machine of the day 2 worker node must be done while the Agent is in the installing phase. Otherwise, the CSR will never be approved and the installation will never complete.

**Review by avishayt — APPROVED** [review]
Your PR looks good - sorry for the noise.  It slipped my mind that we won't set the status to installed until the node is Ready, and that won't happen until the CSR is approved.

**Comment by CrystalChun** [comment]
> Your PR looks good - sorry for the noise. It slipped my mind that we won't set the status to installed until the node is Ready, and that won't happen until the CSR is approved.

No problem! Thank you again for the review!

**Pull request merged**
